ambry-store/src/main/java/com.github.ambry.store/IndexValue.java

```diff
@@ -308,20 +304,21 @@ void setNewSize(long size) {
    * @return the {@link ByteBuffer} representation of this {@link IndexValue}
    */
   ByteBuffer getBytes() {
-    ByteBuffer value = ByteBuffer.allocate(indexValueSizeInBytes);
+    ByteBuffer value = null;
     if (version == PersistentIndex.VERSION_0) {
```

Comment:
Guess you are looking at a older version of the patch. Latest code has switch case only.